### PR TITLE
add-thread-count-fields

### PR DESCRIPTION
**Pull request by praveen-elastic**
Following fields are added in the README: 

sharepoint_sync_thread_count

enterprise_search_sync_thread_count

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -577,6 +577,24 @@ The number of retries to perform when there is a server error. The connector app
 retry_count: 3
 ```
 
+#### `sharepoint_sync_thread_count`
+
+The number of threads the connector will run parallelly for fetching documents from the SharePoint server. By default, the connector uses 5 threads.
```
Comment (leemthompo) [inline README.md:582]:
```suggestion
The number of threads the connector will run in parallel when fetching documents from the SharePoint server. By default, the connector uses 5 threads.
```

File: README.md [hunk 2]
```diff
@@ -577,6 +577,24 @@ The number of retries to perform when there is a server error. The connector app
...
+
+```yaml
+sharepoint_sync_thread_count: 5
+```
+
+#### `enterprise_search_sync_thread_count`
+
+The number of threads the connector will run parallelly for indexing documents to the Enterprise Search instance. By default, the connector uses 5 threads.
```
Comment (leemthompo) [inline README.md:590]:
```suggestion
The number of threads the connector will run in parallel when indexing documents to the Enterprise Search instance. By default, the connector uses 5 threads.
```

File: README.md [hunk 3]
```diff
@@ -577,6 +577,24 @@ The number of retries to perform when there is a server error. The connector app
...
+
+```yaml
+enterprise_search_sync_thread_count: 5
+```
+
+For the Linux distribution with atleast 2 GB RAM and 4 vCPUs, you can increase the thread counts if the overall CPU and RAM are under utilized i.e. below 60-70%.
```
Comment (leemthompo) [inline README.md:596]:
```suggestion
For a Linux distribution with at least 2 GB RAM and 4 vCPUs, you can increase thread counts— if the overall CPU and RAM are underutilized i.e. below 60-70%.
```

**Review by leemthompo — APPROVED** [review]
I'll approve these changes contingent on the rewording and maybe @richkuz can confirm that the thread count info is accurate.

**Comment by serenachou** [comment]
Thanks for the updates @leemthompo  super helpful!

**Pull request merged**